image/templates/helm/shared/templates/_scanner-v4_init.tpl.htpl

```diff
@@ -24,150 +24,112 @@
 {{ $scannerV4Cfg := index . 1 }}
 {{ $_ := false }}
 
+{{/* Sanity check. */}}
+{{- if not (or (eq $.Chart.Name "stackrox-central-services") (eq $.Chart.Name "stackrox-secured-cluster-services")) -}}
+  {{- include "srox.fail" (printf "Unexpected Helm chart name %q." $.Chart.Name) -}}
+{{- end -}}
+
 {{ $componentsCentralChart := dict "indexer" true "matcher" true }}
 {{ $componentsSecuredClusterChart := dict "indexer" true "matcher" false }}
 
-
 {{/* These will be propagated up. */}}
 {{ $components := dict "indexer" false "matcher" false }}
 {{ $dbEnabled  := false }}
 
-{{/* Handle high-level defaulting logic. */}}
 {{- if not $scannerV4Cfg.disable }}
+  {{/* Scanner V4 is switched on. */}}
 
-{{/* Sanity check. */}}
-{{- if not (or (eq $.Chart.Name "stackrox-central-services") (eq $.Chart.Name "stackrox-secured-cluster-services")) -}}
-  {{- include "srox.fail" (printf "Unexpected Helm chart name %q." $.Chart.Name) -}}
-{{- end -}}
-
-{{- if and (kindIs "invalid" $scannerV4Cfg.indexer.disable) (kindIs "invalid" $scannerV4Cfg.matcher.disable) -}}
```

Comment:
We get rid of these component-specific toggles, hence we can simplify this whole if-conditional.